Libraries/pbxbuild/Sources/Phase/ProductTypeResolver.cpp

```diff
@@ -268,6 +269,135 @@ ResolveFrameworkStructure(Phase::Environment const &phaseEnvironment, Phase::Con
     return true;
 }
 
+static ext::optional<std::string>
+FindUmbrellaHeaderName(pbxsetting::Environment const &environment, std::string const &moduleName, pbxproj::PBX::Target::shared_ptr const &target)
+{
+    std::string lowerModuleName;
+    std::transform(moduleName.begin(), moduleName.end(), std::back_inserter(lowerModuleName), ::tolower);
```

Comment:
xcbuild is also intended to run on Linux, which does have a case sensitive filesystem.  My own experiments building on Linux have tripped over stupid header misnames, which I've had to fix.

Since this is something Linux builders would have to worry about anyway, can you make this strict?